apps/timetracker/providers/jira.go

```diff
@@ -0,0 +1,160 @@
+package providers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/andygrunwald/go-jira"
+	"github.com/iReflect/reflect-app/constants"
+
+	"github.com/iReflect/reflect-app/apps/tasktracker"
+	"github.com/iReflect/reflect-app/apps/timetracker"
+	"github.com/iReflect/reflect-app/apps/timetracker/serializers"
+	"github.com/iReflect/reflect-app/libs/utils"
+)
+
+// JiraTimeProvider ...
+type JiraTimeProvider struct {
+}
+
+// JIRAConnection ...
+type JIRAConnection struct {
+	config JIRAConfig
+	client *jira.Client
+}
+
+// JIRAConfig ...
+type JIRAConfig struct {
+	Credentials   tasktracker.Credentials `json:"Credentials"`
```

Comment:
We should make `Credentials` here as well.

Reply:
Done.